doc/architecture/decisions/0021-BLAST-sequence-formats.md

```diff
@@ -0,0 +1,94 @@
+
+## Supported formats
+We support the most common sequence formats for any BLAST program - Fasta format or sequence as plain text
+
+## Decision
+Like other blast search providers we also support 2 ways of input - by pasting sequence in the input box or by uploading a file.
+
+- Input sequence to be accepted in either Fasta format characters or as plain text format (bare sequences). (Sequence IDs may be supported in the future)
+- Sequence should contain minimum 5 characters so that blast can process it properly.
+- No more than 30 sequences can be submitted at a time to prevent overloading JDispatcher.
+- Sequence to be split into 60-character-long lines (Ensembl standard)
+- Will not implement example data for blast as we don't have it on the current site (and no complaints)
+
+## Automatic formatting and validation
+
+For all sequences inserted in the blast input field we do a set of automatic formatting and validation to capture any accidental errors and highlight them on screen.
+
+### Formattings:
+1. Transform all characters to upper-case letters if they aren’t upper-cased.
+2. Remove all spaces (or tabs) between characters
+3. Split the input into 60-character lines (Ensembl standard), except for the header line of FASTA-formatted sequences
+
+| Input type  | Transformations | Support |
+| ------------- | ------------- | ------- |
+| Bare sequence  | None  | Yes |
+| Bare sequence with numbered lines | Accept that a line starting with a digit is valid, and automatically delete the digits. In contrast, treating any digit in the middle of a line is invalid input. | Yes |
+| Bare sequence with empty lines | According to NCBI, blank lines are not allowed in the middle of a bare sequence input. However, we can show leniency and treat empty lines as sequence separators (and split the input into multiple sequences) | Yes |
+| FASTA sequence | Interpret the header line as a separator, and split into multiple sequences if input contains multiple header lines | Yes |
+| FASTA sequence with blank lines | Blank lines are not allowed in the middle of a FASTA input. However, we can show leniency and interpret blank lines as sequence separators. | Yes |
+| A combination of bare sequences and FASTA sequences in one input. | An unlikely scenario, but, technically, can rely on both blank lines and FASTA headers as sequence separators. | Yes |
+| Sequence id | Fetch sequence by its id and insert in the input field | No |
+
+
+## Validation
+For sequence validation we follow NCBI rules (https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)
+### Sequence type detection 
+
+Automatic detection of sequence type has been implemented which determines whether the input sequence is nucleotide or protein as soon as they are entered. Users can always override this selection and run a job. If the input sequence is detected as an amino acid sequence, we change the “Database” to Proteins and select “blastp” as the default Program
```

Comment:
Could use [this file](https://github.com/Ensembl/ensembl-client/blob/0cf19245ef73cd242cf9a6c88b65a435dadddee3/src/content/app/tools/blast/utils/sequenceTypeGuesser.ts) to add more clarity for how we "determine whether the input sequence is nucleotide or protein ":
- First, we remove ambiguous characters from the sequence
- Then, if a sequence has unique characters from amino acid alphabet, we consider it to be a protein
- Otherwise, we count the number of characters that can represent unambiguous nucleotides ("ACGTU"), and if they comprise 90% of the sequence or more, we consider it to be a nucleotide sequence. If not, we consider the sequence to be a protein.